py/desispec/io/emlinefit.py

```diff
@@ -0,0 +1,567 @@
+#!/usr/bin/env python
+
+"""
+desispec.io.emlinefit
+====================
+IO routines for desi_emlinefit_afterburner.
+"""
+
+import os
+import sys
+from astropy.io import fits
+import fitsio
+import numpy as np
+from scipy.optimize import curve_fit
+from desitarget.geomask import match_to
+from desiutil.dust import ext_odonnell, ext_fitzpatrick
+from desiutil.dust import ebv as dust_ebv
+from desiutil.log import get_logger
+from matplotlib.backends.backend_pdf import PdfPages
+import matplotlib.pyplot as plt
+from matplotlib import gridspec
```

Comment:
move these matplotlib imports into the plotting section so that it will only try to import matplotlib if really needed to make a plot (removes a possible fragility from batch processing)

Reply:
done.

Reply:
mmhh.. apparently the related commit failed to pass the checks; I've no idea why.

Reply:
the checks  are complaining about test coverage (new code without tests that actually try that code).  With the new structure separating I/O from algorithms from plotting, that will be a bit easier to add tests.  We'll deal with that a bit later.